src/promptflow/promptflow/azure/operations/_async_run_downloader.py

```diff
@@ -173,11 +173,10 @@ async def _download_single_blob(self, blob_client, local_path: Optional[Path] =
         logger.debug("Downloading blob '%s' to local path '%s'", blob_client.blob_name, local_path.resolve().as_posix())
         local_path.parent.mkdir(parents=True, exist_ok=True)
         async with blob_client:
-            with open(local_path, "wb") as f:
-                stream = await blob_client.download_blob()
-                data = await stream.readall()
-                # TODO: File IO may block the event loop, consider using thread pool. e.g. to_thread() method
-                f.write(data)
+            stream = await blob_client.download_blob()
```

Comment:
will this fail OOM if the downloaded file is too large?

Reply:
Good catch, it indeed will cause OOM error if the blob is too large. I have modified this part to stream download the content,